controllers/clusteroperator.go

```diff
@@ -56,6 +56,12 @@ func relatedObjects() []osconfigv1.ObjectReference {
 			Resource: "namespaces",
 			Name:     ComponentNamespace,
 		},
+		{
+			Group:     "metal3.io",
+			Resource:  "baremetalhosts",
+			Name:      "baremetalhosts.metal3.io",
```

Comment:
This doesn't look right to me. There's unlikely to be a BareMetalHost named `baremetalhosts.metal3.io`, and even if there were there's no reason it should be a related object?
I presume the intention here was to reference the baremetalhost **CRD**, in which case I think the Group should be `apiextensions.k8s.io` and the Resource should be `CustomResourceDefinition`.

Reply:
@kirankt pointed out that the [equivalent code in the MAO](https://github.com/openshift/machine-api-operator/blob/71aeab4ff840991c8569baf3817e8640e03a2dfe/pkg/operator/status.go#L202-L207) just has the name as an empty string. Presumably that means that _all_ `baremetalhosts` would be treated as related, and that was the intention here (i.e. not to mark the CRD as related). I think we should drop the `Name`.